File: C++ Assignment/011.cpp
#include <iostream>
using namespace std;

void generateTriangle(int n)
{
    char letter = 65;
    int j = 1;
    for (int i = 0; i < n; i++)
    {
        // spacing
        for (int s = n - i; s > 0; s--)
        {
            cout << " ";
        }

        // left skewed triangle
        for (int c = 0; c <= (i + j) / 2; c++)
        {
            cout << letter++;
        }

        letter--;

        // right skewed triangle
        for (int c = 0; c < (i + j) / 2; c++)
        {
            cout << --letter;
        }
        cout << endl;

        letter = 65;
        j++;
    }
}

int main()
{
    int n;
    cout << "Enter a number: ";
    cin >> n;

    generateTriangle(n);

    return 0;
}

File: Python Assignment/011.py
def generateTriangle(num: int) -> None:
    
    char: str = 65;
    j: int = 1;
    for i in range(num):
        for s in range(num - i, 0, -1):
            print(" ", end="")
        
        for c in range(((i+j)//2)+1):
            print(chr(char), end="")
            char += 1
        
        char -= 2
        
        for c in range((i+j)//2):
            print(chr(char), end="")
            char -= 1
            
        print(end="\n")
        char = 65;
        j += 1


num = int(input("Enter a number: "))
generateTriangle(num)